src/publisher/index.ts

```diff
@@ -264,9 +256,23 @@ export class Publisher {
   constructSpan(message: PubsubMessage): Span | undefined {
     const spanAttributes = {
       data: message.data,
+      // Add Opentelemetry semantic convention attributes to the span
+      [MessagingAttribute.MESSAGING_TEMP_DESTINATION]: false,
+      [MessagingAttribute.MESSAGING_SYSTEM]: 'pubsub',
+      [MessagingAttribute.MESSAGING_OPERATION]: '', // operation expected to be empty
+      [MessagingAttribute.MESSAGING_DESTINATION]: this.topic.name,
+      [MessagingAttribute.MESSAGING_DESTINATION_KIND]: 'topic',
+      [MessagingAttribute.MESSAGING_MESSAGE_ID]: message.messageId,
+      [MessagingAttribute.MESSAGING_PROTOCOL]: 'pubsub',
+      [MessagingAttribute.MESSAGING_MESSAGE_PAYLOAD_COMPRESSED_SIZE_BYTES]: 0,
+      [MessagingAttribute.MESSAGING_MESSAGE_PAYLOAD_SIZE_BYTES]:
+        message.data?.length,
```

Comment:
nit given this check above https://github.com/googleapis/nodejs-pubsub/blob/78f31e4aa637a6ca755627ad3c54cbac59040e39/src/publisher/index.ts#L157

i was a bit confused otherwise if this was just string length 
```suggestion
        (message.data as Buffer).length,
```

Reply:
That does seem a little bit more clear, but I could see it either way. If you did want to make that change, might want something like:

`(message.data as Buffer?)?.length`

...which is looking a bit messy, so maybe the existing one ends up being nicer after all.

Reply:
I accepted Aaron suggestion :)